@navikt/core/tokens/darkside/figma/plugin/AkselVariablesInterface.ts

```diff
@@ -0,0 +1,258 @@
+  private remoteConfigURL =
+    "https://cdn.nav.no/designsystem/@navikt/tokens/figma-config.json";
+
+  constructor() {
+    const config = _config as FigmaTokenConfig;
+    this.config = config;
+    this.meta = { timestamp: config.timestamp, version: config.version };
+    this.Figma = new FigmaPluginInterface();
+  }
+
+  /**
+   * Because of "dynamic-page" loading in Figma,
+   * we need to initialize the plugin trough async methods.
+   * This method should be called before any other methods.
+   */
+  async init(): Promise<void> {
+    await this.Figma.init();
+  }
+
+  async useRemoteConfig(): Promise<void> {
+    const newConfig = await fetch(this.remoteConfigURL)
+      .then((res) => res.json())
+      .catch((err) => {
+        console.error(err);
+        throw new Error("Error fetching config from CDN 😱");
+      });
+    this.meta = { timestamp: newConfig.timestamp, version: newConfig.version };
+
+    this.config = newConfig;
+  }
+
+  updateVariables(): void {
+    this.updateGlobalColorCollection(this.config.colors.light.global);
+    this.updateGlobalColorCollection(this.config.colors.dark.global);
+    this.updateSemanticColorCollection(this.config.colors);
+    this.updateScaleCollection(this.config.radius);
+    this.updateScaleCollection(this.config.spacing);
+    console.info("Updated variables!");
+  }
+
+  private updateGlobalColorCollection(
+    globalColorTheme: FigmaConfigEntry,
+  ): void {
+    let collection = this.Figma.getCollection(globalColorTheme.name);
+
+    if (!collection) {
+      collection = this.Figma.createCollection(globalColorTheme.name);
+    }
+
+    /**
+     * Correctly sorts the token-scale for global colors
+     * 000 - 100 - 200... etc.
+     */
+    const sortedTokens = globalColorTheme.tokens.sort((a, b) => {
+      const getLastNumber = (name: string) => {
+        const matches = name.match(/\d+/g);
+        return matches ? parseInt(matches[matches.length - 1], 10) : 0;
+      };
+      return getLastNumber(a.name) - getLastNumber(b.name);
+    });
+
+    for (const token of sortedTokens) {
+      /* Color values can only be defined by strings */
+      if (typeof token.value !== "string") {
+        throw new Error(`Token value is not a string: ${token}`);
+      }
+
+      let variable = this.Figma.getVariable(token.name, collection.id);
+
+      if (!variable) {
+        variable = this.Figma.createVariable(
+          token.name,
+          collection,
+          token.figmaType,
+        );
+      }
+
+      this.Figma.setVariableValue(
+        variable,
+        figma.util.rgba(token.value),
+        collection.defaultModeId,
+      );
+
+      this.Figma.setVariableMetadata(variable, {
+        codeSyntax: token.code,
+        description: token.comment ?? "",
+        hiddenFromPublishing: collection.hiddenFromPublishing,
+        scopes: token.scopes,
+      });
+    }
+
+    console.info("Updated collection: ", collection.name);
```

Comment:
Not a big deal, but I think a space will be added automatically
```suggestion
    console.info("Updated collection:", collection.name);
```

Reply:
TIL
![Screenshot 2024-10-03 at 15 16 40](https://github.com/user-attachments/assets/99c66d8d-5cd2-423a-917a-e144de511904)